### ISSUE TITLE
"Loading the module for the configuration unit failed." When using components

### ISSUE THREAD

**Issue by ShannyMan** [microsoft/VisualStudioDSC#14]
I'm attempting to install and configure Visual Studio using the `Components` option versus the `VSConfigFile`. When I run the winget configure, I get the below error:

_"Loading the module for the configuration unit failed."_

![image](https://github.com/microsoft/VisualStudioDSC/assets/29736122/89393d1a-ef31-4505-88c9-00411f5ec8ea)

Here is my yaml:
```
properties:
  configurationVersion: 0.2.0

  resources:

    - resource: Microsoft.WinGet.DSC/WinGetPackage
      directives:
        description: Visual Studio Code
      settings:
        id: Microsoft.VisualStudioCode
        source: winget

    - resource: Microsoft.WinGet.DSC/WinGetPackage
      id: vsPackage
      directives:
        description: Visual Studio 2022 Community Edition
      settings:
        id: Microsoft.VisualStudio.2022.Community
        source: winget

    - resource: Microsoft.VisualStudio.DSC/VSComponents
      dependsOn:
        - vsPackage
      directives:
        description: Install required VS Workloads
      settings:
        productId: Microsoft.VisualStudio.Product.Community
        channelId: VisualStudio.17.Release
        includeRecommended: true
        components:
          - Microsoft.VisualStudio.Workload.Azure
          - Microsoft.VisualStudio.Workload.ManagedDesktop
          - Microsoft.VisualStudio.Workload.Data
          - Microsoft.VisualStudio.Workload.NetCrossPlat
          - Microsoft.VisualStudio.Workload.NetWeb
          - Microsoft.VisualStudio.Workload.Universal
```

Is there something I'm missing or doing wrong?

Thank you!

**Comment by Preecington** [comment]
Thanks for reporting this! We'll need a little more information to diagnose what is going on here.

Doing the following will help us get the information we need:

1. Run winget configure with the `--verbose` and `--open-logs` arguments. This will open a folder with WinGet logs - zip these files into a compressed file.
2. Download and run the [VS Log Collection Tool](https://aka.ms/vscollect.exe). This will generate a `vslogs.zip` file in your `%tmp%` directory.
3. Open Visual Studio Installer and select Report a problem from the top right
![](https://developercommunity.visualstudio.com/images/ReportProblemVSInstaller.png)
4. On the webpage that opens, point to this GitHub issue in the description, and attach the WinGet logs zip and the `vslogs.zip` file, then submit the ticket.

Thanks for taking the time to help us understand this issue better!

**Comment by parkerreno** [comment]
I had this error and it was because I was not running winget as admin.  Once I ran it in an elevated terminal, it worked.

**Comment by ShannyMan** [comment]
Thanks for getting back to me. I figured out my issue. I hadn't installed the module. 🤦 Sorry, still a bit new to winget and there doesn't seem to be a lot of great support and docs out there for using configuration files so far, so it's been a journey.

When I was first running this script, I was doing it via Dev Home. Dev Home automatically initialized winget and the other `Microsoft.WinGet.DSC` module, so I was a bit oblivious to what was happening behind the scenes.

I hope you don't mind, but I submitted a PR to add some very basic installation instructions, linking to the PowerShell Gallery.
